applications/ChimeraApplication/python_scripts/fluid_chimera_analysis.py

```diff
@@ -0,0 +1,158 @@
+
+        # Import parallel modules if needed
+        # has to be done before the base-class constuctor is called (in which the solver is constructed)
+        if (parameters["problem_data"]["parallel_type"].GetString() == "MPI"):
+            raise Exception("MPI-Chimera is not implemented yet")
+
+        super(FluidChimeraAnalysis,self).__init__(model,self.full_parameters)
+
+    def Initialize(self):
+        super(FluidChimeraAnalysis,self).Initialize()
+        self.__SetChimeraInternalPartsFlag()
+
+    def _CreateProcesses(self, parameter_name, initialization_order):
+
+        list_of_processes = super(FluidChimeraAnalysis,self)._CreateProcesses(parameter_name, initialization_order)
+
+        main_model_part = self.model[self.fluid_parameters["model_part_name"].GetString()]
+        domain_size = main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        solver_type = self.fluid_parameters["solver_type"].GetString()
+
+
+        # Creating the necessary variant of the apply chimera process.
+        if domain_size == 2:
+            if(solver_type == "Monolithic" or solver_type == "monolithic" or solver_type == "chimera_embedded"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessMonolithic2d(main_model_part,self.chimera_levels)
+            elif (solver_type == "fractional_step" or solver_type == "FractionalStep"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessFractionalStep2d(main_model_part,self.chimera_levels)
+        else:
+            if(solver_type == "Monolithic" or solver_type == "monolithic" or solver_type == "chimera_embedded"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessMonolithic3d(main_model_part,self.chimera_levels)
+            elif (solver_type == "fractional_step" or solver_type == "FractionalStep"):
+                self.chimera_process = KratosChimera.ApplyChimeraProcessFractionalStep3d(main_model_part,self.chimera_levels)
+
+        self.chimera_process.SetEchoLevel(self.chimera_echo_lvl)
+        self.chimera_process.SetReformulateEveryStep(self.reformulate_every_step)
+
+        return list_of_processes
+
+    def __SetChimeraInternalPartsFlag(self):
+        '''
+            This function sects the flag CHIMERA_INTERNAL_BOUNDARY on the specified modelparts
+            so that they are excluded from the extract surface operation later on.
+        '''
+        for mp_name in self.chimera_internal_parts:
+            KratosMultiphysics.VariableUtils().SetFlag(KratosChimera.CHIMERA_INTERNAL_BOUNDARY, True,  self.model[mp_name].Nodes)
+
+    def InitializeSolutionStep(self):
+        self.PrintAnalysisStageProgressInformation()
+        self.ApplyBoundaryConditions() #here the processes are called
+        self.ChangeMaterialProperties() #this is normally empty
+        ## The following will construct the constraints
+        self.chimera_process.ExecuteInitializeSolutionStep()
```

Comment:
I think this should be encapsulated in the Chimera solver. In this way the the AnalysisStage would be almost or even completely identical to the FluidDynamics base one.

Reply:
I suppose it can now be done ! Previously it was not the case. Anyway I will see how it can be done !

Reply:
The only annoying thing then would be there will be a `process` in solver !

Reply:
Now these check and initialize and finalize of the process are done in the solvers.

Reply:
Cool. As I told you in person, I took this same approach for the level set correction in the embedded solvers as it gives you much more control of what you want to do in there.